rest/blip_sync.go

```diff
@@ -631,24 +631,17 @@ func (bh *blipHandler) sendRevAsDelta(sender *blip.Sender, seq db.SequenceID, do
 		}
 	}
 
-	bh.Logf(base.LevelDebug, base.KeySync, "DELTA: getting oldBody docID: %s - deltaSrcRevID: %v", base.UD(docID), deltaSrcRevID)
-	oldBody, err := bh.db.GetRev(docID, deltaSrcRevID, true, nil)
-	if err != nil {
-		bh.Logf(base.LevelDebug, base.KeySync, "DELTA: couldn't get oldBody - falling back to full body replication... err: %v", err)
-		// fall back to standard full-body replication if we can't do a delta
-		bh.sendRevOrNorev(sender, seq, docID, revID, knownRevs, maxHistory)
-		return
-	}
-
-	// generate delta
-	delta, err := base.Diff(oldBody, newBody)
-	if err != nil {
-		bh.Logf(base.LevelWarn, base.KeySync, "DELTA: couldn't generate delta - falling back to full body replication... err: %v", err)
+	delta, err := bh.db.GetDelta(docID, deltaSrcRevID, revID)
+	if err != nil || delta == nil {
```

Comment:
double nesting of ifs is a bit awkward, could it be `if err != nil {} else if delta == nil {}`  ?